api/src/repositories/draft-repository.ts

```diff
@@ -0,0 +1,121 @@
+import { QueryResult } from 'pg';
+import SQL, { SQLStatement } from 'sql-template-strings';
+import { z } from 'zod';
+import { ApiExecuteSQLError } from '../errors/api-error';
+import { BaseRepository } from './base-repository';
+
+export const WebformDraftList = z.object({
+  webform_draft_id: z.number(),
+  name: z.string(),
+  data: z.object({}),
+  create_date: z.date(),
+  update_date: z.date().nullable()
+});
+
+export type WebformDraftList = z.infer<typeof WebformDraftList>;
+
+/**
+ * A repository class for accessing draft data.
+ *
+ * @export
+ * @class DraftRepository
+ * @extends {BaseRepository}
+ */
+export class DraftRepository extends BaseRepository {
+  async deleteDraft(draftId: number): Promise<QueryResult> {
+    const sqlStatement = SQL`
+      DELETE from webform_draft
+      WHERE webform_draft_id = ${draftId};
+    `;
+
+    const response = await this.connection.sql(sqlStatement);
+
+    if (!response) {
+      throw new ApiExecuteSQLError('Failed to delete draft', [
+        'ProjectRepository->deleteDraft',
+        'response was null or undefined, expected response != null'
+      ]);
+    }
+
+    return response;
+  }
+
+  async getSingleDraft(draftId: number): Promise<{ id: number; name: string; data: any }> {
```

Comment:
Rather than resolving type `{ id: number; name: string; data: any }`, we could make this a `SELECT *` statement, and return `Promise<WebformDraft>` instead